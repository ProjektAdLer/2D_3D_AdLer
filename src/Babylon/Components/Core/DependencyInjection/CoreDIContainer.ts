import { Container } from "inversify";
import IPresentation from "../Presentation/API/IPresentation";
import Presentation from "../Presentation/API/Presentation";
import IBusinessLogic from "../Presentation/API/IBusinessLogic";
import BusinessLogic from "../BusinessLogic/API/BusinessLogic";
import IDataAccess from "./../BusinessLogic/API/IDataAccess";
import DataAccess from "../DataAccess/API/DataAccess";
import ICore from "../API/ICore";
import Core from "../API/Core";
import IEngineManager from "../Presentation/EngineManager/IEngineManager";
import EngineManager from "../Presentation/EngineManager/EngineManager";
import ISceneView from "../Presentation/SceneManager/ISceneView";
import SceneView from "../Presentation/SceneManager/SceneView";
import SceneViewModel from "../Presentation/SceneManager/SceneViewModel";

import CORE_TYPES from "./types";
import ScenePresenter from "../Presentation/SceneManager/ScenePresenter";

var CoreDIContainer = new Container();

CoreDIContainer.bind<IPresentation>(CORE_TYPES.IPresentation)
  .to(Presentation)
  .inSingletonScope();
CoreDIContainer.bind<IBusinessLogic>(CORE_TYPES.IBusinessLogic)
  .to(BusinessLogic)
  .inSingletonScope();
CoreDIContainer.bind<IDataAccess>(CORE_TYPES.IDataAccess)
  .to(DataAccess)
  .inSingletonScope();
CoreDIContainer.bind<ICore>(CORE_TYPES.ICore).to(Core).inSingletonScope();
CoreDIContainer.bind<IEngineManager>(CORE_TYPES.IEngineManager)
  .to(EngineManager)
  .inSingletonScope();
CoreDIContainer.bind<ISceneView>(CORE_TYPES.ISceneView)
  .to(SceneView)
  .inSingletonScope();
CoreDIContainer.bind<SceneViewModel>(SceneViewModel)
  .toSelf()
  .inSingletonScope();
CoreDIContainer.bind<ScenePresenter>(ScenePresenter).toSelf();

export default CoreDIContainer;
