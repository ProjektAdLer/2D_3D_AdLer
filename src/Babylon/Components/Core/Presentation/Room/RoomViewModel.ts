import { injectable } from "inversify";
import { ROOMSIZE } from "../../BusinessLogic/RoomConfigurator/RoomConfigurator";

@injectable()
export default class RoomViewModel {
  private roomSize: ROOMSIZE;

  get RoomSize(): ROOMSIZE {
    if (!this.roomSize) throw new Error("roomSize not found!");
    return this.roomSize;
  }

  set RoomSize(newRoomSize: ROOMSIZE) {
    this.roomSize = newRoomSize;
  }
}
