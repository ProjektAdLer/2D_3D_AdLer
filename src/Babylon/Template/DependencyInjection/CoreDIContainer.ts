import { BusinessLogic } from "./../BusinessLogic/API/BusinessLogic";
import { Container } from "inversify";
import { PresentationLogic } from "../PresentationLogic/API/PresentationLogic";
import { DataAccess } from "../DataAccess/API/DataAccess";

// rename this container when using the template
var CoreDIContainer = new Container();

CoreDIContainer.bind<PresentationLogic>(PresentationLogic).toSelf();
CoreDIContainer.bind<BusinessLogic>(BusinessLogic).toSelf();
CoreDIContainer.bind<DataAccess>(DataAccess).toSelf();

export default CoreDIContainer;
