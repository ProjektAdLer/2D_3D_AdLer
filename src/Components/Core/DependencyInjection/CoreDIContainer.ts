import { LoadWorldController } from "../Presentation/React/LoadWorldButton/LoadWorldController";
import ILoadWorldController from "../Presentation/React/LoadWorldButton/ILoadWorldController";
import { Container } from "inversify";
import CORE_TYPES from "./CoreTypes";
import ISceneView from "../Presentation/Babylon/SceneManagement/ISceneView";
import SceneView from "../Presentation/Babylon/SceneManagement/SceneView";
import SceneViewModel from "../Presentation/Babylon/SceneManagement/SceneViewModel";
import ScenePresenter from "../Presentation/Babylon/SceneManagement/ScenePresenter";
import MainScene from "../Presentation/Babylon/SceneManagement/MainScene";
import ICreateSceneClass from "../Presentation/Babylon/SceneManagement/ICreateSceneClass";
import IScenePresenter from "../Presentation/Babylon/SceneManagement/IScenePresenter";
import infrastructureDIContainer from "./InfrastructureDIContainer";
import useCaseDIContainer from "./UseCases/UseCaseDIConatiner";
import PortsDIContainer from "./Ports/PortsDIContainer";
import BuilderDIContainer from "./Builders/BuilderDIContainer";
import INavigation from "../Presentation/Babylon/Navigation/INavigation";
import Navigation from "../Presentation/Babylon/Navigation/Navigation";
import NavigationConfiguration from "../Presentation/Babylon/Navigation/NavigationConfiguration";

var CoreDIContainer = new Container();

// Scene
CoreDIContainer.bind<ISceneView>(CORE_TYPES.ISceneView)
  .to(SceneView)
  .inSingletonScope();
CoreDIContainer.bind(SceneViewModel).toSelf().inSingletonScope();
CoreDIContainer.bind<IScenePresenter>(CORE_TYPES.IScenePresenter)
  .to(ScenePresenter)
  .inSingletonScope();

// Navigation
CoreDIContainer.bind<INavigation>(CORE_TYPES.INavigation)
  .to(Navigation)
  .inSingletonScope();
CoreDIContainer.bind(NavigationConfiguration).toSelf();

// Controllers
CoreDIContainer.bind<ILoadWorldController>(CORE_TYPES.ILoadWorldController)
  .to(LoadWorldController)
  .inSingletonScope();

// bind other CreateSceneClass here for testing puposes -MK
CoreDIContainer.bind<ICreateSceneClass>(CORE_TYPES.ICreateSceneClass).to(
  MainScene
);

CoreDIContainer.load(infrastructureDIContainer);
CoreDIContainer.load(useCaseDIContainer);
CoreDIContainer.load(PortsDIContainer);
CoreDIContainer.load(BuilderDIContainer);

export default CoreDIContainer;
