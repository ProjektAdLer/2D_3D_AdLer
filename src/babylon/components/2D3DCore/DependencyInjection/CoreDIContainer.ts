import { Container } from 'inversify';
import { PresentationLogic } from '../Presentation/API/PresentationLogic';
import { SBWViewModel } from '../Presentation/SchoenerBunterWuerfel/SBWViewModel';
import { BusinessLogic } from '../BusinessLogic/API/BusinessLogic';
import { Calculator } from '../BusinessLogic/Calculator';
import { DataAccess } from '../DataAccess/API/DataAccess';

var CoreDIContainer = new Container();
CoreDIContainer.bind<PresentationLogic>(PresentationLogic).toSelf();
CoreDIContainer.bind<SBWViewModel>(SBWViewModel).toSelf();
CoreDIContainer.bind<BusinessLogic>(BusinessLogic).toSelf();
CoreDIContainer.bind<Calculator>(Calculator).toSelf();
CoreDIContainer.bind<DataAccess>(DataAccess).toSelf();

export default CoreDIContainer;
