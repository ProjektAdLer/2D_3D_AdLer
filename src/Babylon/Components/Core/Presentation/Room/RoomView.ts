import { MeshBuilder, Scene, Vector3 } from "@babylonjs/core";
import { injectable, inject } from "inversify";
import IRoomView from "./IRoomView";
import RoomViewModel from "./RoomViewModel";
import { ROOMSIZE } from "../../BusinessLogic/RoomConfigurator/RoomConfigurator";

@injectable()
export default class RoomView implements IRoomView {
  private roomViewModel: RoomViewModel;
  private roomScale: number;
  constructor(@inject(RoomViewModel) roomViewModel: RoomViewModel) {
    this.roomViewModel = roomViewModel;
  }
  setRoomScale(roomSize: ROOMSIZE): number {
    if (roomSize === ROOMSIZE.Small) return 1;
    else if (roomSize === ROOMSIZE.Medium) return 2;
    else if (roomSize === ROOMSIZE.Large) return 3;
    else throw new Error("Invalid roomSize to scale Room!");
  }
  createFloor(scene: Scene, roomScale: number): void {
    MeshBuilder.CreatePlane(
      "Floor",
      {
        width: 5 * roomScale,
        height: 3.5 * roomScale,
      },
      scene
    );
  }

  createWalls(scene: Scene, roomScale: number): void {
    MeshBuilder.CreatePlane(
      "Northwall",
      {
        width: 5 * roomScale,
        height: 3,
      },
      scene
    )
      .rotate(new Vector3(1, 0, 0), Math.PI / 2)
      .translate(new Vector3(0, 0, 1), 3.5);
    MeshBuilder.CreatePlane(
      "Eastwall",
      {
        width: 3.5 * roomScale,
        height: 3,
      },
      scene
    )
      .rotate(new Vector3(0, 1, 0), Math.PI / 2)
      .rotate(new Vector3(0, 0, 1), Math.PI / 2)
      .translate(new Vector3(1, 0, 0), 5);
    MeshBuilder.CreatePlane(
      "Southwall",
      {
        width: 5 * roomScale,
        height: 3,
      },
      scene
    )
      .rotate(new Vector3(1, 0, 0), -Math.PI / 2)
      .rotate(new Vector3(0, 0, 1), -Math.PI / 2);
    MeshBuilder.CreatePlane(
      "Westwall",
      {
        width: 3.5 * roomScale,
        height: 3,
      },
      scene
    )
      .rotate(new Vector3(0, 1, 0), -Math.PI / 2)
      .rotate(new Vector3(0, 0, 1), -Math.PI / 2)
      .translate(new Vector3(1, 0, 0), -5);
  }
}
