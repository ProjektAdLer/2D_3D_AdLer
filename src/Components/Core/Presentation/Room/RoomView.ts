import {
  Scene,
  Mesh,
  VertexData,
  StandardMaterial,
  Texture,
} from "@babylonjs/core";
import { injectable } from "inversify";
import IRoomView from "./IRoomView";
import RoomViewModel from "./RoomViewModel";
import floorTexture from "../../../../Assets/wooden_floor.png";

@injectable()
export default class RoomView implements IRoomView {
  private viewModel: RoomViewModel;

  set ViewModel(newViewModel: RoomViewModel) {
    this.viewModel = newViewModel;
  }

  public displayRoom(): void {
    this.createWalls();
    this.createFloor();
  }

  private createFloor(): void {
    if (!this.viewModel)
      throw new Error(
        "ViewModel not set. Use the ViewModel setter before calling this method"
      );

    var floorIndices = this.createFloorIndices();
    var floorPositions = this.createFloorPositions();

    var floorMesh = new Mesh("Floor", this.viewModel.scene.Value);
    var normals = [] as number[];
    var uvs = [0, 1, 0, 0, 1, 0, 1, 1];
    VertexData.ComputeNormals(floorPositions, floorIndices, normals);
    var vertexData = new VertexData();
    vertexData.positions = floorPositions;
    vertexData.indices = floorIndices;
    vertexData.normals = normals;
    vertexData.uvs = uvs;
    vertexData.applyToMesh(floorMesh);

    var floorMaterial = new StandardMaterial(
      "floorMaterial",
      this.viewModel.scene.Value
    );
    floorMaterial.diffuseTexture = new Texture(
      floorTexture,
      this.viewModel.scene.Value
    );
    floorMesh.material = floorMaterial;
  }

  private createWalls(): void {
    if (!this.viewModel)
      throw new Error(
        "ViewModel not set. Use the ViewModel setter before calling this method"
      );

    var positions = this.createRoomPositions();
    var wallIndices = this.createWallIndices();

    var wallMesh = new Mesh("Walls", this.viewModel.scene.Value);
    var normals = [] as number[];
    VertexData.ComputeNormals(positions, wallIndices, normals);
    var vertexData = new VertexData();
    vertexData.positions = positions;
    vertexData.indices = wallIndices;
    vertexData.normals = normals;
    vertexData.applyToMesh(wallMesh);
    wallMesh.convertToFlatShadedMesh();

    var wallMaterial = new StandardMaterial(
      "wallMaterial",
      this.viewModel.scene.Value
    );
    wallMaterial.diffuseColor = this.viewModel.wallColor.Value;
    wallMesh.material = wallMaterial;
  }

  private createFloorPositions() {
    var roomWidth = this.viewModel.roomWidth.Value / 2;
    var roomLength = this.viewModel.roomLength.Value / 2;
    var baseHeight = this.viewModel.baseHeight.Value;
    var wallThickness = this.viewModel.wallThickness.Value;
    return [
      roomWidth + wallThickness,
      baseHeight,
      roomLength + wallThickness, // Same as: 44 north east outer floor
      -roomWidth - wallThickness,
      baseHeight,
      roomLength + wallThickness, // Same as: 49 north west outer floor
      -roomWidth - wallThickness,
      baseHeight,
      -roomLength - wallThickness, // Same as: 54 south west outer floor
      roomWidth + wallThickness,
      baseHeight,
      -roomLength - wallThickness, // Same as: 59 south east outer floor
    ];
  }

  private createRoomPositions() {
    var roomWidth = this.viewModel.roomWidth.Value / 2;
    var roomLength = this.viewModel.roomLength.Value / 2;
    var baseHeight = this.viewModel.baseHeight.Value;
    var roomHeight = this.viewModel.roomHeight.Value;
    var doorWidth = this.viewModel.doorWidth.Value / 2;
    var doorHeight = this.viewModel.doorHeight.Value;
    var wallThickness = this.viewModel.wallThickness.Value;
    return [
      roomWidth,
      baseHeight,
      roomLength, // 0 north east, ground
      roomWidth,
      baseHeight,
      -roomLength, // 1 south east, ground
      -roomWidth,
      baseHeight,
      -roomLength, // 2 south west, ground
      -roomWidth,
      baseHeight,
      roomLength, // 3 north west, ground
      roomWidth,
      baseHeight + roomHeight,
      roomLength, // 4 north east, ceiling
      doorWidth,
      baseHeight,
      roomLength, // 5 northern floor, east doorpoint
      doorWidth,
      baseHeight + doorHeight,
      roomLength, // 6 northern wall, east doorpoint
      doorWidth,
      baseHeight + roomHeight,
      roomLength, // 7 northern ceiling, east doorpoint
      -doorWidth,
      baseHeight,
      roomLength, // 8 northern floor, west doorpoint
      -doorWidth,
      baseHeight + doorHeight,
      roomLength, // 9 northern wall, west doorpoint
      -doorWidth,
      baseHeight + roomHeight,
      roomLength, // 10 northern ceiling, west doorpoint
      -roomWidth,
      baseHeight + roomHeight,
      roomLength, // 11 north west ceiling
      -roomWidth,
      baseHeight,
      doorWidth, // 12 western floor, north doorpoint
      -roomWidth,
      baseHeight + doorHeight,
      doorWidth, // 13 western wall, north doorpoint
      -roomWidth,
      baseHeight + roomHeight,
      doorWidth, // 14 western ceiling, north doorpoint
      -roomWidth,
      baseHeight,
      -doorWidth, // 15 western floor, south doorpoint
      -roomWidth,
      baseHeight + doorHeight,
      -doorWidth, // 16 western wall, south doorpoint
      -roomWidth,
      baseHeight + roomHeight,
      -doorWidth, // 17 western ceiling, south doorpoint
      -roomWidth,
      baseHeight + roomHeight,
      -roomLength, // 18 south west ceiling
      -doorWidth,
      baseHeight,
      -roomLength, // 19 southern floor, west doorpoint
      -doorWidth,
      baseHeight + doorHeight,
      -roomLength, // 20 southern wall, west doorpoint
      -doorWidth,
      baseHeight + roomHeight,
      -roomLength, // 21 southern ceiling, west doorpoint
      doorWidth,
      baseHeight,
      -roomLength, // 22 southern floor, east doorpoint
      doorWidth,
      baseHeight + doorHeight,
      -roomLength, // 23 southern wall, east doorpoint
      doorWidth,
      baseHeight + roomHeight,
      -roomLength, // 24 southern ceiling, east doorpoint
      roomWidth,
      baseHeight + roomHeight,
      -roomLength, // 25 south east ceiling
      roomWidth,
      baseHeight,
      -doorWidth, // 26 eastern floor, south doorpoint
      roomWidth,
      baseHeight + doorHeight,
      -doorWidth, // 27 eastern wall, south doorpoint
      roomWidth,
      baseHeight + roomHeight,
      -doorWidth, // 28 eastern ceiling, south doorpoint
      roomWidth,
      baseHeight,
      doorWidth, // 29 eastern floor, north doorpoint
      roomWidth,
      baseHeight + doorHeight,
      doorWidth, // 30 eastern wall, north doorpoint
      roomWidth,
      baseHeight + roomHeight,
      doorWidth, // 31 eastern ceiling, north doorpoint

      roomWidth + wallThickness,
      baseHeight + roomHeight,
      roomLength + wallThickness, // 32 north east outer ceiling
      doorWidth,
      baseHeight + roomHeight,
      roomLength + wallThickness, // 33 northern outer ceiling, east doorpoint
      -doorWidth,
      baseHeight + roomHeight,
      roomLength + wallThickness, // 34 northern outer ceiling, west doorpoint
      -roomWidth - wallThickness,
      baseHeight + roomHeight,
      roomLength + wallThickness, // 35 north west outer ceiling
      -roomWidth - wallThickness,
      baseHeight + roomHeight,
      doorWidth, // 36 western outer ceiling, north doorpoint
      -roomWidth - wallThickness,
      baseHeight + roomHeight,
      -doorWidth, // 37 western outer ceiling, south doorpoint
      -roomWidth - wallThickness,
      baseHeight + roomHeight,
      -roomLength - wallThickness, // 38 south west outer ceiling
      -doorWidth,
      baseHeight + roomHeight,
      -roomLength - wallThickness, // 39 southern outer ceiling, west doorpoint
      doorWidth,
      baseHeight + roomHeight,
      -roomLength - wallThickness, // 40 southern outer ceiling, east doorpoint
      roomWidth + wallThickness,
      baseHeight + roomHeight,
      -roomLength - wallThickness, // 41 south east outer ceiling
      roomWidth + wallThickness,
      baseHeight + roomHeight,
      -doorWidth, // 42 eastern outer ceiling, south doorpoint
      roomWidth + wallThickness,
      baseHeight + roomHeight,
      doorWidth, // 43 eastern outer ceiling, north doorpoint
      roomWidth + wallThickness,
      baseHeight,
      roomLength + wallThickness, // 44 north east outer floor
      doorWidth,
      baseHeight,
      roomLength + wallThickness, // 45 north outer floor, east doorpost
      doorWidth,
      baseHeight + doorHeight,
      roomLength + wallThickness, // 46 north outer wall, east doorpost
      -doorWidth,
      baseHeight + doorHeight,
      roomLength + wallThickness, // 47 north outer wall, west doorpost
      -doorWidth,
      baseHeight,
      roomLength + wallThickness, // 48 north outer floor, west doorpost
      -roomWidth - wallThickness,
      baseHeight,
      roomLength + wallThickness, // 49 north west outer floor
      -roomWidth - wallThickness,
      baseHeight,
      doorWidth, // 50 western outer floor, north doorpost
      -roomWidth - wallThickness,
      baseHeight + doorHeight,
      doorWidth, // 51 western outer wall, north doorpost
      -roomWidth - wallThickness,
      baseHeight + doorHeight,
      -doorWidth, // 52 western outer wall, south doorpost
      -roomWidth - wallThickness,
      baseHeight,
      -doorWidth, // 53 western outer floor, south doorpost
      -roomWidth - wallThickness,
      baseHeight,
      -roomLength - wallThickness, // 54 south west outer floor
      -doorWidth,
      baseHeight,
      -roomLength - wallThickness, // 55 southern outer floor, west doorpost
      -doorWidth,
      baseHeight + doorHeight,
      -roomLength - wallThickness, // 56 southern outer wall, west doorpost
      doorWidth,
      baseHeight + doorHeight,
      -roomLength - wallThickness, // 57 southern outer wall, east doorpost
      doorWidth,
      baseHeight,
      -roomLength - wallThickness, // 58 southern outer floor, east doorpost
      roomWidth + wallThickness,
      baseHeight,
      -roomLength - wallThickness, // 59 south east outer floor
      roomWidth + wallThickness,
      baseHeight,
      -doorWidth, // 60, eastern outer floor, south doorpost
      roomWidth + wallThickness,
      baseHeight + doorHeight,
      -doorWidth, // 61, eastern outer wall, south doorpost
      roomWidth + wallThickness,
      baseHeight + doorHeight,
      doorWidth, // 62, eastern outer wall, north doorpost
      roomWidth + wallThickness,
      baseHeight,
      doorWidth, // 63, eastern outer floor, north doorpost
    ];
  }

  private createFloorIndices() {
    return [0, 1, 2, 2, 3, 0];
  }

  private createWallIndices() {
    return [
      0,
      4,
      5,
      4,
      7,
      5,
      5,
      7,
      8,
      7,
      10,
      8,
      8,
      10,
      11,
      8,
      11,
      3, // northern Wall
      3,
      11,
      12,
      12,
      11,
      14,
      14,
      15,
      12,
      14,
      17,
      15,
      17,
      2,
      15,
      17,
      18,
      2, // western wall
      2,
      21,
      19,
      2,
      18,
      21,
      21,
      24,
      19,
      19,
      24,
      22,
      22,
      24,
      25,
      25,
      1,
      22, // southern wall
      1,
      25,
      28,
      1,
      28,
      26,
      26,
      28,
      29,
      29,
      28,
      31,
      29,
      31,
      0,
      0,
      31,
      4, // eastern wall
      32,
      35,
      11,
      32,
      11,
      4, // northern wall thickness
      35,
      38,
      18,
      35,
      18,
      11, // western wall thickness
      38,
      41,
      25,
      38,
      25,
      18, // southern wall thickness
      41,
      32,
      4,
      41,
      4,
      25, // eastern wall thickness
      44,
      45,
      32,
      45,
      33,
      32,
      45,
      48,
      33,
      48,
      34,
      33,
      49,
      34,
      48,
      49,
      35,
      34, // northern outer wall
      49,
      50,
      35,
      50,
      36,
      35,
      50,
      53,
      36,
      53,
      37,
      36,
      53,
      54,
      37,
      54,
      38,
      37, // western outer wall
      38,
      54,
      55,
      55,
      39,
      38,
      55,
      58,
      39,
      58,
      40,
      39,
      58,
      59,
      40,
      59,
      41,
      40, // southern outer wall
      59,
      60,
      41,
      60,
      42,
      41,
      60,
      43,
      42,
      60,
      63,
      43,
      63,
      44,
      43,
      44,
      32,
      43, // eastern outer wall
    ];
  }
}
