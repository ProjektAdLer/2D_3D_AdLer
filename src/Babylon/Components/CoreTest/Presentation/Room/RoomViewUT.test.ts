import { Scene, Engine } from "@babylonjs/core";
import CoreDIContainer from "../../../Core/DependencyInjection/CoreDIContainer";
import CORE_TYPES from "../../../Core/DependencyInjection/types";
import RoomView from "../../../Core/Presentation/Room/RoomView";
import RoomViewModel from "../../../Core/Presentation/Room/RoomViewModel";

jest.mock("@babylonjs/core");

describe("RoomView", () => {
  let roomView: RoomView;
  let roomViewModel: RoomViewModel;

  beforeEach(() => {
    roomView = CoreDIContainer.get(CORE_TYPES.IRoomView);
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  test("CreateFloor throws Error if ViewModel is not set.", () => {
    expect(() => {
      roomView.createFloor(
        new Scene(new Engine(document.createElement("canvas")))
      );
    }).toThrowError();
  });

  test("CreateWalls throws Error if ViewModel is not set.", () => {
    expect(() => {
      roomView.createWalls(
        new Scene(new Engine(document.createElement("canvas")))
      );
    }).toThrowError();
  });

  test("View Model is being Set by the Setter", () => {
    roomView.ViewModel = CoreDIContainer.get(RoomViewModel);
    //@ts-ignore
    expect(roomView).toMatchSnapshot({
      viewModel: expect.any(RoomViewModel),
    });
  });
});
