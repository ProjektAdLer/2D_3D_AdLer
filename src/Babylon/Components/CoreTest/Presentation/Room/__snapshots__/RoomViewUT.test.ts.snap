// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomView View Model is being Set by the Setter 1`] = `
Object {
  "viewModel": Any<RoomViewModel>,
}
`;
